test/test.cpp

```diff
@@ -11,3 +12,68 @@ TEST(basic, basic_test) {
   // Assert
   ASSERT_EQ(5, c);
 }
+TEST(graph, check_connection) {
+  std::vector<int> vec = {1, 2, 3, 4};
+  Graph graph(5);
+  Layer a1(0, 1);
+  Layer a2(1, 2);
+  Layer a3(2, 1);
+  Layer a4(3, 2);
+  graph.input(a1, 3, vec);
```

Comment:
please use verbs for action - e.g `setInput(a1, vec)`

Reply:
done